crates/interpreter/src/valid.rs

```diff
@@ -412,24 +406,32 @@ struct Expr<'a, 'm> {
     is_body: bool,
     locals: Vec<ValType>,
     labels: Vec<Label<'m>>,
+    side_table: Vec<Option<SideTableEntryView>>,
+}
+
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq)]
+struct SideTableBranch<'m> {
+    parser: &'m [u8],
+    side_table_len: usize,
```

Comment:
This is not the `len`, this is the index of the entry, which just happens to be the length before pushing it.

```suggestion
    side_table: usize,
```